src/hammer-vlsi/defaults.yml

```diff
@@ -312,6 +312,25 @@ vlsi.inputs:
     #  - y (int) - The Y coordinate of the bump assigned to this net
     assignments: []
 
+  # Pin placement mode. (str)
+  # Specifies how to arrange pins for your design.
+  # Valid options:
+  # none - This mode lets the CAD tool do whatever it wants (providing no constraints).
+  #   Typically this is sane but unpredictable.
+  # generated - Use the assignments list below
+  # auto - (not implemented yet) looks at the hierarchy and guess where pins should go
+  pin_mode: none
+  pin:
+    # List of PinAssignment Structs.
+    #   You must specify pins and one of either macro or layers and side.
+    # - pins (str) - The name(s) of pins that will be assigned.
+    # - side (Optional[str]) - The side of the chip/block these pins will be placed along.
```

Comment:
High level question: how do I control the width/height of the generated pins? And their x/y locations? e.g. let's say we use a floorplanning DSL or other method to floorplan and want to control the physical locations.

Reply:
You don't and shouldn't control the width or the height. These are intended to be conneted by abutment, not via drop.

The x/y locations will come later (I can't recall if @colinschmidt included that in a comment but we discussed offline).

Reply:
What's the usecase for this then? A case I encounter is using P&R tools to build a macro that will be ingested at some higher level, and it is critical that we are able to control the width, height, x, and y of the pin. I think even the underlying TCL command support this...

Reply:
Of course the TCL commands support it; I'm saying you don't need it for this level of API. For a bottom (vertical route) pin, for example, you don't care about the pin height at all. The width is the width of a single route. The actual x position does matter, but not much, so we've decided there are more important issues to tackle than to implement that now.

Reply:
How hard would it be to add support for "width" (maybe "depth" but it is less important) and positioning? I think if these two were present this API would become a lot more useful.

Reply:
You can always write TCL if you need something very specific. I think that width is probably less useful than you seem to be implying. The only reasons I see for adding width are:
1. If you want to do analog routing, but IMO that should inherit its width from the non-default routing rule used to draw it.
2. If you want to do via-drop pins, but I question the usefulness of this outside of a few niches.

I don't think positioning matters enough to gate this PR, but I'll let @colinschmidt speak to that. If you have a specific use case that might help motivate this, please share it.

Reply:
It's not hard but I think we have different idea of what this API is for.
The primary use case for this for users like me is to be able to control where the pins of inner digital blocks are.
For example, if I have a block that will be placed adjacent to the chip boundary I can't have any pins going out on that side, so I would constrain the pins to be on other sides.
If the pins are so fixed I presume they would be defined by some IP's LEF and so you could use the `macro` portion of the API to force the inner pins to be outer pins.
In general, digital place and route does not require such specific placement. LEFs are very robust in terms of pin placement definitions so exporting you block as such should allow for easy ingestion in all tools I'm familiar with.

Reply:
Deferred to #371